src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/__tests__/ExposureSubmissionUITests.swift

```diff
@@ -311,11 +315,9 @@ class ENAUITests_04_ExposureSubmissionUITests: XCTestCase {
 		// Click secondary button to skip symptoms screens and immediately go to warn others screen.
 		XCTAssertTrue(app.buttons["AppStrings.ExposureSubmission.secondaryButton"].waitForExistence(timeout: .medium))
 		app.buttons["AppStrings.ExposureSubmission.secondaryButton"].tap()
-
-		app.alerts["Sie wollen keine Warnung verschicken?"].buttons["Nicht warnen"].tap()
-		
-		waitForNonExistence(of: app.buttons["AppStrings.ExposureSubmission.primaryButton"], timeout: .long)
```

Comment:
Is this dismissed by `addUIInterruptionMonitor(withDescription: "System Dialog")`? How do we distinguish between two different alert actions? Which one is triggered?

Reply:
This way of using `addUIInterruptionMonitor` by just returning `false` triggers the default implementation which uses the `.default` action on the alert. We can tweak the handler, make our own logic, e.g. by using identifiers to find the right button on the right alert and return `true` as in 'yes, we got this covered'.

I consider this monitor as a general handler for all kind of non-deterministic interruptions and try to keep it simple. If I know for sure that a certain alert will trigger AND I need to handle that alert in a very particular way as in `test_ThankYouScreen_withWarnOthers` I target the (expected) alert directly and handle it in place.

There is [another resource on this topic](https://useyourloaf.com/blog/handling-system-alerts-in-ui-tests/) and we can cover more questions in the dev meeting if needed as this feature is generally really cool to handle privacy prompts.